pkg/gitopsjob/job.go

```diff
@@ -0,0 +1,240 @@
+							ImagePullPolicy: corev1.PullAlways,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("10m"),
+									corev1.ResourceMemory: resource.MustParse("50Mi"),
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "OPERATION",
+									Value: gitopsConfig.OperationType.String(),
+								},
+								{
+									Name:  "REPOURL",
+									Value: gitopsConfig.RepositoryURL,
+								},
+								{
+									Name:  "RESOURCE",
+									Value: gitopsConfig.ResourceName,
+								},
+								{
+									Name:  "BRANCH",
+									Value: gitopsConfig.Branch,
+								},
+								{
+									Name:  "CONTEXT",
+									Value: gitopsConfig.Context,
+								},
+								{
+									Name:  "NAMESPACE",
+									Value: resourceNamespace,
+								},
+								{
+									Name: "GITHUB_TOKEN",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: "has-github-token",
+											},
+											Key: "token",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err := client.Create(ctx, &gitopsJob)
+	return err
+}
+
+// WaitForJob waits for the given Kubernetes job to complete.
+// If it errors out, or the given timeout is reached, an error is returned
+func WaitForJob(log logr.Logger, ctx context.Context, client ctrlclient.Client, clientset kubernetes.Interface, jobName string, jobNamespace string, timeout time.Duration) error {
+	var job batchv1.Job
+	var err error
+	for stay, timeout := true, time.After(timeout); stay; {
+		err = client.Get(context.Background(), types.NamespacedName{Namespace: jobNamespace, Name: jobName}, &job)
+		if err != nil {
+			// If the error is anything but a isnotfound error, return the error
+			// If the resource wasn't found, keep trying up to the timeout, in case the job hasn't appeared yet
+			if !k8sErrors.IsNotFound(err) {
+				return err
+			}
+		}
+
+		// The CompletionTime in the job's status will only get set when the Job completes successfully, so check for its presence
+		if job.Status.CompletionTime != nil {
+			return nil
+		}
+		if job.Status.Failed >= 5 {
+			// Attempt to get the Pod logs
+			errMsg, err := GetPodLogs(context.Background(), client, clientset, jobName, jobNamespace)
+			if err != nil {
+				log.Error(err, "unable to retrieve pod logs for job "+jobName)
+				return fmt.Errorf("gitops generation job failed")
+			} else {
+				return fmt.Errorf(errMsg)
+			}
+		}
+		time.Sleep(1 * time.Second)
+		select {
+		case <-timeout:
+			stay = false
+		default:
+		}
+	}
+
+	if err != nil && k8sErrors.IsNotFound(err) {
+		return fmt.Errorf("gitops generation job was not found after timeout reached")
+	} else if err != nil {
+		return err
+	}
+
+	return fmt.Errorf("gitops generation job did not complete in time")
```

Comment:
```suggestion
	if err != nil && k8sErrors.IsNotFound(err) {
		return fmt.Errorf("gitops generation job was not found after timeout reached")
	} 

	return err
```

I think we can just do